internal/condition/metrics/metrics.go

```diff
@@ -0,0 +1,73 @@
+// Packge metrics provides various metric and telemetry definitions for OpenFGA Conditions.
+package metrics
+
+import (
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+
+	"github.com/openfga/openfga/internal/build"
+	"github.com/openfga/openfga/internal/server/config"
+	"github.com/openfga/openfga/internal/utils"
+)
+
+const (
+	subsystem = "conditions"
+)
+
+// Metrics provides access to Condition metrics.
+var Metrics *ConditionMetrics
+
+func init() {
+	m := &ConditionMetrics{
+		compilationTime: promauto.NewHistogram(prometheus.HistogramOpts{
+			Namespace: build.ProjectName,
+			Subsystem: subsystem,
+			Name:      "condition_compilation_duration_ms",
+			Help:      "A histogram measuring the compilation time (in milliseconds) of a Condition.",
+			Buckets:   []float64{1, 5, 15, 50, 100, 250, 500, 1000},
+		}),
+
+		evaluationTime: promauto.NewHistogram(prometheus.HistogramOpts{
+			Namespace: build.ProjectName,
+			Subsystem: subsystem,
+			Name:      "condition_evaluation_duration_ms",
+			Help:      "A histogram measuring the evaluation time (in milliseconds) of a Condition.",
+			Buckets:   []float64{0.1, 0.25, 0.5, 1, 5, 15, 50, 100, 250, 500},
+		}),
+
+		evaluationCost: promauto.NewHistogram(prometheus.HistogramOpts{
+			Namespace:                       build.ProjectName,
+			Name:                            "condition_evaluation_cost",
+			Help:                            "A histogram of the CEL evaluation cost of a Condition in a Relationship Tuple",
+			Buckets:                         utils.LinearBuckets(0, config.DefaultMaxConditionEvaluationCost, 10),
+			NativeHistogramBucketFactor:     1.1,
+			NativeHistogramMaxBucketNumber:  config.DefaultMaxConditionEvaluationCost,
+			NativeHistogramMinResetDuration: time.Hour,
+		}),
+	}
+
+	Metrics = m
```

Comment:
Do we intend this to be singleton pattern?  If so, should we only overwrite Metrics if it is nil?

Reply:
`init` is called once upon package initialization, so the intent is the `Metrics` variable gets assigned once at package init, which as you point out is a singleton.

We have similar usage patterns in `internal/condition/condition.go`.

Reply:
In this case, should we only set Metrics only if Metrics is nil?  Otherwise, someone accidentally calling init more than once will cause unexpected behaviour?

Reply:
`init` is a function that gets called automatically, it's not exported so you can't call it